tensorflow/contrib/layers/python/layers/layers_test.py

```diff
@@ -2553,6 +2641,18 @@ def testNHWCAndNCHWTrainingProduceSameOutput(self):
       self.assertAllClose(nhwc, nchw, atol=1e-4, rtol=1e-4)
 
 
+class BatchNormFloat16Test(BatchNormTest):
+  dtype = dtypes.float16
+
+
+class BatchNormFloat32Test(BatchNormTest):
+  dtype = dtypes.float32
+
+
+class BatchNormFloat64Test(BatchNormTest):
+  dtype = dtypes.float64
+
+
 class LayerNormTest(test.TestCase):
```

Comment:
Let's take that test itself and run for float16, then, leaving the others untouched?